packages/components/forms/src/text-input/TextInput.mdx

```diff
@@ -0,0 +1,78 @@
+---
+title: 'TextInput'
+type: 'component'
+status: 'stable'
+slug: /components/forms/src/text-input/
+github: 'https://github.com/contentful/forma-36/tree/master/packages/components/forms/src/text-input'
+storybook: 'https://f36-storybook.contentful.com/?path=/story/form-elements-textinput--basic'
+typescript: ./TextInput.tsx
+---
+
+`TextInput` is a form component, that allows users to input text or numbers and should be used in a form context.
+
+## Table of contents
+
+- [How to use TextInput](#how-to-use-TextInput)
+- [Component variations](#component-variations)
+- [Content recommendations](#content-recommendations)
+- [Accessibility](#accessibility)
+- [Props](#props)
+
+## How to use TextInput
+
+The TextInput component can be configured in a number of different ways. Use this guidance to determine the correct variation based on the context.
+
+TextInput is supporting different types of inputs, like `'text' | 'email' | 'file' | 'number' | 'password' | 'search' | 'url'`, where `text` is a default.
+Label property is always required and passed to the `<input />` as `aria-label`.
+
+## Code examples
+
+```jsx
+<React.Fragment>
+  <TextInput type="email" name="email" label="Email address" />
+</React.Fragment>
+```
+
+## TextInput variations
+
+### Character count (`countCharacters`)
+
+The TextInput can be configured to display a character count. This is useful when an input field has a character limit.<br/>
+**Note**: The `maxLength` property must be also provided.
+
+```jsx
+<TextInput
+  type="text"
+  label="name"
+  name="name"
+  id="input-1"
+  placeholder="Type your name"
+  countCharacters
+  maxLength={150}
+/>
+```
+
+### TextInput with copy button (`withCopyButton`)
+
+We provide a variant of the TextInput that is a composed with CopyButton. In order to use it add `withCopyButton` property to your implementation.
+
+```jsx
+<TextInput label="Token" name="token" id="input-2" withCopyButton />
+```
+
+## Content recommendations
+
+- Use direct, succinct copy that helps the user to successfully complete the form
+- Do not preface the instructions with introductory text, such as "please"
+- To add additional context, link out to documentation
```

Comment:
I didn't understand this, what does it mean?

Reply:
Me neither, hahaha